src/definition/api/IApi.ts

```diff
@@ -0,0 +1,82 @@
+import { IHttp, IModify, IPersistence, IRead } from '../accessors';
+import { IApiExample } from './IApiExample';
+import { IApiEndpointInfo } from './IEndpoint';
+import { IApiRequest } from './IRequest';
+import { IApiResponse } from './IResponse';
+
+export interface IApiEndpoint {
```

Comment:
I am still of the opinion this should be a class. That way we can have helper methods for converting various things from one to another. For example, the `IApiRequest` has `content: any`. But let's be real, most people are going to expect that to be a json value with a format of `{ [key: string]: string }`. So, we could have helper methods for the TypeScript to assume that something is what they expect it to be. Also, for converting from a string to json. Could also have helper methods for common things such as a successful response `return this.success()` or `return this.json(statusCode: number, data: object)`.

Reply:
I see your points, would it be better to have a class in place of the interface or a class that implements the interface named `ApiEndpoint`?

Reply:
Now that's a good question, probably doesn't matter too much. Thinking about it though, might be best to just have one so there is less source for confusion when an external developer tries to look into the code to see how to do things.